src/rrd_cgi.c

```diff
@@ -560,13 +559,13 @@ static char *rrdsetenv(
     const char **args)
 {
     if (argc >= 2) {
-        const size_t len = strlen(args[0]) + strlen(args[1]) + 2;
-        char *xyz = (char *) malloc(len);
+        char *xyz = NULL;
 
+        asprintf(&xyz, "%s=%s", args[0], args[1]);
```

Comment:
noo :(.  I'll leave it to some *other* brave soul to suggest adding a _second_ copy of snprintf.c to the project.

There's some other places that use a stack buffer for formatting error messages, so I copied that instead.